File: Pharmacy.Core/CriteriaObjects/Bases/PagedCriteriaObject.cs
namespace Pharmacy.Core.CriteriaObjects.Bases
{
    public class PagedCriteriaObject : CriteriaObject
    {
        public int CurrentPage { get; set; }
        public int TotalCountOfPages { get; set; }
        public int TotalCountOfRecords { get; set; }
        public int RecordsCountOfPerPage { get; set; }
        public int PageOfStart { get; set; }
        public int PageOfEnd { get; set; }
        public string? SearchKey { get; set; }
        public string? Where { get; set; }
        public object? Parameter { get; set; }
    }
}

File: Pharmacy.Client/src/app/contracts/criteria/base/paged-criteria-object.ts
import { CriteriaObject } from "./criteria-object";

export class PagedCriteriaObject extends CriteriaObject{
    currentPage?:number;
    totalCountOfPages?:number;
    totalCountOfRecords?:number;
    recordsCountOfPerPage?:number;
    pageOfStart?:number;
    pageOfEnd?:number;
    searchKey?:string;
    where?:number;
    parameter?:object;
}